tools/twix/src/panels/parameter.rs

```diff
@@ -82,7 +87,27 @@ impl Widget for &mut ParameterPanel {
                         }
                     }
                 });
+                ui.add_enabled_ui(settable, |ui| {
+                    if ui.button("Save to disk").clicked() {
+                        if let Some(address) = self.nao.get_address() {
+                            match serde_json::from_str::<Value>(&self.parameter_value) {
+                                Ok(value) => {
+                                    if let Err(error) = self
+                                        .repository_parameters
+                                        .write(&address, &self.path, &value)
+                                    {
+                                        error!("Failed to write value to repository: {error:#?}");
+                                    }
+                                }
+                                Err(error) => {
+                                    error!("Failed to serialize parameter value: {error:#?}")
+                                }
```

Comment:
I would like to have some sort of indicator in the UI showing whether the most recent save operation succeeded or not.
Either show the full error message or some shorter message referring the user to the log file.

The set button doesn't do that either... 😞

Reply:
Some sort of a status indicator label?

Reply:
I thought a bit about this, We can implement such a thing for the UI.
But when I think of your remark of not blocking the UI, it becomes a bit tricky IMO.

If I understood correctly, to do non-blocking UI, we can attach a closure which has a binding to the UI element or a state variable of the panel and that closure is run inside the repository_parameters' runner. Do you think such a setup is acceptable?

Reply:
I created an issue to cover this topic, #216 . I would prefer to complete this PR in the current scope and extend so that this feature is available on the main branch sooner than later.